src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -249,7 +250,7 @@ class AuthScreens extends React.Component {
         if (this.unsubscribeGroupShortcut) {
             this.unsubscribeGroupShortcut();
         }
-        NetworkConnection.stopListeningForReconnect();
+        cleanupSession();
```

Comment:
NAB: I was wondering why do we do this here instead of in `clearStorageAndRedirect`. I feels like the clean up logic is spread and it could be more centralized.

Reply:
Ah, mainly because we are setting up the subscriptions in `AuthScreens` `componentDidMount()` here: 

https://github.com/Expensify/App/blob/ccef3d903dc3affe7ddd85aa5d3166930f714899/src/libs/Navigation/AppNavigator/AuthScreens.js#L157-L159

so I thought it made more sense to tear them down in `componentWillUnmount()`.

We definitely could do it in `clearStorageAndRedirect()` but I think making the "cleanup" stuff a side effect of losing the `authToken` is OK because we set up a lot of logic in `componentDidMount()` that could also be triggered by the presence of the `authToken`.

Basically the assumption is that `AuthScreens` mounted = `we have authToken` and `AuthScreen` unmount = `we lost the authToken`.

But yeah I wouldn't mind cleaning this up eventually or doing something differently if it becomes complicated enough to where people are thoroughly confused about how it works.

Reply:
>Ah, mainly because we are setting up the subscriptions in AuthScreens componentDidMount() here:

Ohh that makes sense, I only looked at the code in `componentWillUnmount`

>But yeah I wouldn't mind cleaning this up eventually or doing something differently if it becomes complicated enough to where people are thoroughly confused about how it works.

Agreed, I think it is good for now like it is, there is a good enough reason to keep it there.